src/Illuminate/Foundation/EnvironmentDetector.php

```diff
@@ -48,7 +48,10 @@ protected function detectConsoleEnvironment(Closure $callback, array $args)
         // and if it was that automatically overrides as the environment. Otherwise, we
         // will check the environment as a "web" request like a typical HTTP request.
         if (! is_null($value = $this->getEnvironmentArgument($args))) {
-            return head(array_slice(explode('=', $value), 1));
+            $env = head(array_slice(explode('=', $value), 1));
+            putenv("APP_ENV=$env");
```

Comment:
I was trying to run a command with the option --env
but I had a condition within the command as follows
```php
if (env('APP_ENV') == 'testing')
```
Clearly I had to change the line to:
```php
if (app()->enviroment ('local'))
```

```php
if ($this->laravel->environment ('local'))
```
Why env('APP_ENV') had the value **local**